File: frontend/types/stats.ts
export interface CountryCount {
  code: string;
  count: number;
}

export interface HotspotRegion {
  region: string;
  count: number;
  max_severity?: string;
  severe_or_extreme_count?: number;
}

export interface TrendAnomaly {
  dimension: string;
  key: string;
  current_count: number;
  rolling_avg: number;
  ratio: number;
}

export interface Stats {
  active_count: number;
  global_risk_score: number;
  score_breakdown: Record<string, unknown>;
  by_country: Record<string, number>;
  by_category: Record<string, number>;
  by_severity: Record<string, number>;
  by_source: Record<string, number>;
  top_countries: CountryCount[];
  hotspot_regions: HotspotRegion[];
  trend_anomalies: TrendAnomaly[];
  last_ingest: string | null;
  canonical_event_count?: number;
  observed_event_count?: number;
}

File: backend/app/schemas/stats.py
"""Stats API schemas."""

from datetime import datetime

from pydantic import BaseModel


class CountryCount(BaseModel):
    code: str
    count: int


class HotspotRegion(BaseModel):
    region: str
    count: int


class StatsResponse(BaseModel):
    active_count: int
    global_risk_score: int = 0
    score_breakdown: dict[str, float | int] = {}
    by_country: dict[str, int] = {}
    by_category: dict[str, int] = {}
    by_severity: dict[str, int] = {}
    top_countries: list[CountryCount] = []
    hotspot_regions: list[HotspotRegion] = []
    last_ingest: datetime | None = None
